### DISCUSSION TITLE
Are we allowed to train a machine learning model using whisper's output?

### DISCUSSION THREAD

**Discussion by dani-local** [openai/whisper#1393] [General]
Let's say I want to train a speech to text model for which I have some audio files with no human transcripts, am I allowed to use whisper to generate transcripts I can train my model on?

**Comment by glangford** [comment]
There is no issue using whisper, the more important question is what rights you have to use the audio files.

**Comment by dani-local** [comment]
according to OpenAI's terms of use https://openai.com/policies/terms-of-use 

> These Terms of Use apply when you use the services of OpenAI, L.L.C. or our affiliates, including our application programming interface, software, tools, developer services, data, documentation, and websites (“Services”). 

> (c) Restrictions. You may not (i) use the Services in a way that infringes, misappropriates or violates any person’s rights; (ii) reverse assemble, reverse compile, decompile, translate or otherwise attempt to discover the source code or underlying components of models, algorithms, and systems of the Services (except to the extent such restrictions are contrary to applicable law); **(iii) use output from the Services to develop models that compete with OpenAI; **

**(iii) use output from the Services to develop models that compete with OpenAI;**

I'm not sure if those terms of use apply to this project, to whisper and the models released here. I hope somebody from OpenAI can comment on this.

**Reply by ryanheise** [reply]
Only the OpenAI services are governed by those terms of use, while this GitHub repository has a separate open source license:

https://github.com/openai/whisper/blob/main/LICENSE

So as @glangford said, it is a matter of whether you have the rights to use the particular audio files you want to use.

**Reply by glangford** [reply]
Good clarification, thank you @ryanheise.

**Comment by dani-local** [comment]
@ryanheise, regarding you comment: 

> Only the OpenAI services are governed by those terms of use, while this GitHub repository has a separate open source license:

I'm not so sure about that, see how "Services", note the uppercase, are defined in the terms-of-use https://openai.com/policies/terms-of-use. See below, with "Services" they seem to include " application programming interface, software, tools, developer services, data, documentation, and websites".

> These Terms of Use apply when you use the services of OpenAI, L.L.C. or our affiliates, including our application programming interface, software, tools, developer services, data, documentation, and websites (“Services”).

I would like somebody from OpenAI to comment on this.

**Reply by ryanheise** [reply]
The README page of this repository tells you exactly which license applies, and it's as simple as that.

You don't need to bother OpenAI's (presumably busy) legal department for legal interpretation, you just need to read GitHub's terms of service which are also binding on OpenAI and any other person or company that chooses to release software on this platform as open source. Since you have signed up with a GitHub account yourself, presumably you have read and understood and accepted GitHub's terms of service, and have understood how GitHub's terms pertain to licensing, whether YOU license something or whether OpenAI does it. You agreed to be bound by the same terms that OpenAI themselves are bound by, and so you would be expected to understand exactly how licenses apply to open source projects hosted by anyone here on GitHub.

> with "Services" they seem to include " application programming interface

The Whisper program may be another kind of API, although it is not a "Service".

So there are two different things here which you are confusing:

1. The OpenAI API (an online paid-for "Service") available at https://openai.com/blog/openai-api
2. The Whisper program (an offline open source "Program") available at https://github.com/openai/whisper.

Each one comes with its own license terms, and you should read the terms relevant to the one you're using. The terms you quoted mention "Service", so it must refer to (1) since that's the only one that's a Service.

> I would like somebody from OpenAI to comment on this.

OpenAI [say it on their website](https://openai.com/research/whisper):

> > We’ve trained and are open-sourcing a neural net called Whisper that approaches human level robustness and accuracy on English speech recognition. [View code](https://github.com/openai/whisper)

Following that link takes you a README page that tells you exactly which open source license applies. Read that README and what it says about the license for more details, in conjunction with reviewing the GitHub terms of service that you have already agreed to.
